src/test/javascript/spec/integration/code-editor/code-editor-container.integration.spec.ts

```diff
@@ -226,66 +212,66 @@ describe('CodeEditorContainerIntegration', () => {
         containerFixture.detectChanges();
 
         // container
-        expect(container.commitState).to.equal(CommitState.CLEAN);
-        expect(container.editorState).to.equal(EditorState.CLEAN);
-        expect(container.buildOutput.isBuilding).to.be.false;
-        expect(container.unsavedFiles).to.be.empty;
+        expect(container.commitState).toBe(CommitState.CLEAN);
+        expect(container.editorState).toBe(EditorState.CLEAN);
+        expect(container.buildOutput.isBuilding).toBe(false);
+        expect(container.unsavedFiles).toStrictEqual({});
 
         // file browser
-        expect(checkIfRepositoryIsCleanStub).to.have.been.calledOnce;
-        expect(getRepositoryContentStub).to.have.been.calledOnce;
-        expect(container.fileBrowser.errorFiles).to.deep.equal(extractedErrorFiles);
-        expect(container.fileBrowser.unsavedFiles).to.be.empty;
+        expect(checkIfRepositoryIsCleanStub).toHaveBeenCalledTimes(1);
+        expect(getRepositoryContentStub).toHaveBeenCalledTimes(1);
+        expect(container.fileBrowser.errorFiles).toEqual(extractedErrorFiles);
+        expect(container.fileBrowser.unsavedFiles).toBeArrayOfSize(0);
 
         // ace editor
-        expect(container.aceEditor.isLoading).to.be.false;
-        expect(container.aceEditor.commitState).to.equal(CommitState.CLEAN);
+        expect(container.aceEditor.isLoading).toBe(false);
+        expect(container.aceEditor.commitState).toBe(CommitState.CLEAN);
 
         // actions
-        expect(container.actions.commitState).to.equal(CommitState.CLEAN);
-        expect(container.actions.editorState).to.equal(EditorState.CLEAN);
-        expect(container.actions.isBuilding).to.be.false;
+        expect(container.actions.commitState).toBe(CommitState.CLEAN);
+        expect(container.actions.editorState).toBe(EditorState.CLEAN);
+        expect(container.actions.isBuilding).toBe(false);
 
         // status
-        expect(container.fileBrowser.status.commitState).to.equal(CommitState.CLEAN);
-        expect(container.fileBrowser.status.editorState).to.equal(EditorState.CLEAN);
+        expect(container.fileBrowser.status.commitState).toBe(CommitState.CLEAN);
+        expect(container.fileBrowser.status.editorState).toBe(EditorState.CLEAN);
 
         // build output
-        expect(getBuildLogsStub).to.have.been.calledOnce;
-        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).to.deep.equal(extractedBuildLogErrors);
-        expect(container.buildOutput.isBuilding).to.be.false;
+        expect(getBuildLogsStub).toHaveBeenCalledTimes(1);
+        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).toIncludeSameMembers(extractedBuildLogErrors);
```

Comment:
Is the order of extracted errors random? If not, then a regular `toEqual` should probably work as well.

Reply:
toEqual doesn't work as for some reason jest compares two different array types and fails. By just checking the members this is not an issue. I'll add a comment.
I couldn't resolve the issue.

Reply:
I think I figured it out: `BuildLogEntryArray.extractErrors` uses `return this.map()…`. Therefore, the return type ends up being a `BuildLogEntryArray` that now contains `Annotations` instead of `BuildLogEntry`, thus directly (and unexpectedly to users of the return value) violating the extends definition of `BuildLogEntryArray`.

Changing the code in `extractErrors` to `return Array.from(this.map(…)…)` would make sure the types are correct and enable the `toEqual` assertion. I think this would be a good code quality improvement not only here in the test but for all callers of `extractErrors`.